File: Python/02_storage.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# sudo pip install blessings (for color terminal output)
import os 
from couchbase.client import Couchbase
from blessings import Terminal

t = Terminal()

os.system('clear')
print t.bold_red("--------------------------------------------------------------------------")
print t.bold_red("Couchbase Storage Operations")
print t.bold_red("--------------------------------------------------------------------------")
print

# establish connection
couchbase = Couchbase("127.0.0.1:8091", "default", "")

# connect to default bucket
cb = couchbase["default"]

print t.bold("Set a Key-Value")
print
# create a key-value pair, set("key", ttl, flags, value)
cb.set("mytest", 0, 0, 1)

print t.bold("Retrieve Key-Value")
print
print "cb.get(\"mytest\")[2] = " + str(cb.get("mytest")[2])
print



print t.bold("Add key that exists already")
print
try:
	# this raises key exists exception, uncomment to see error
	cb.add("mytest", 0, 0, 2)
except 	Exception as e:
	print 'Exception Raised:', e.args[0]#, type(e), e
	print


print t.bold("Replace key-value with a new one")
print
# success
cb.replace("mytest", 0, 0, 2)
print "cb.get(\"mytest\")[2] = " + str(cb.get("mytest")[2])
print


print t.bold("Try to replace doc that doesn't exist")
print
try:
	# this raises key exists exception, uncomment to see error
	cb.replace("mytest4", 0, 0, 4) 
except 	Exception as e:
	print 'Exception Raised:', e.args[0]
	print
	


print t.bold("Add another key-value pair")
print
# success
cb.add("mytest3", 0, 0, 3)
print "cb.get(\"mytest3\")[2] = " + str(cb.get("mytest3")[2])
print


cb.delete("mytest")
cb.delete("mytest3")

File: PHP/02_storage.php
<?php

echo "--------------------------------------------------------------------------\n";
echo "\tCouchbase Storage Operations\n";
echo "--------------------------------------------------------------------------\n";


$cb = new Couchbase();

// Create a key
$cb->set("mytest", 1);
echo("\n--------------\n");
echo("Get mytest :");
echo( $cb->get("mytest") );


// try to add a key that exists
$result = $cb->add("mytest", 2);
if ($result) {
	echo("\n--------------\n");
	echo("Add succeeded");
} else {
	echo("\n--------------\n");
	echo("Add failed : key exists");	
}


// success
$result = $cb->replace("mytest", 2);
echo("\n--------------\n");
echo("Get mytest :");
echo( $cb->get("mytest") );


// try to replace a key that does not exist
$result = $cb->add("mytest4", 2);
if ($result) {
	echo("\n--------------\n");
	echo("Replace succeeded");
} else {
	echo("\n--------------\n");
	echo("Replace failed : key does not exist");	
}

$cb->delete("mytest");

$result = $cb->delete("mytest3");
if ($result) {
	echo("\n--------------\n");
	echo("Delete succeeded");
} else {
	echo("\n--------------\n");
	echo("Delete failed : key does not exist");	
}

echo("\n--------------\n");

?>